From b98b278e008a3374dfc79906fdf8302eae4646d5 Mon Sep 17 00:00:00 2001
From: Tomas Zak <tomas.zak@turris.com>
Date: Thu, 30 Jan 2025 20:02:23 +0100
Subject: [PATCH] python-bcrpyt: fix compiling error AtomicI64 for powerpc

Upgrade pyo3 cargo crate to version 0.21.0 to avoid error
no `AtomicI64` in `sync::atomic`
---
 .../001-fix-compiling-pyo3-powerpc.patch      | 64 +++++++++++++++++++
 1 file changed, 64 insertions(+)
 create mode 100644 lang/python/python-bcrypt/patches/001-fix-compiling-pyo3-powerpc.patch

diff --git a/lang/python/python-bcrypt/patches/001-fix-compiling-pyo3-powerpc.patch b/lang/python/python-bcrypt/patches/001-fix-compiling-pyo3-powerpc.patch
new file mode 100644
index 000000000..bf0d82c58
--- /dev/null
+++ b/lang/python/python-bcrypt/patches/001-fix-compiling-pyo3-powerpc.patch
@@ -0,0 +1,64 @@
+Index: bcrypt-4.1.2/src/_bcrypt/Cargo.lock
+===================================================================
+--- bcrypt-4.1.2.orig/src/_bcrypt/Cargo.lock
++++ bcrypt-4.1.2/src/_bcrypt/Cargo.lock
+@@ -243,9 +243,9 @@ dependencies = [
+ 
+ [[package]]
+ name = "pyo3"
+-version = "0.20.0"
++version = "0.21.0"
+ source = "registry+https://github.com/rust-lang/crates.io-index"
+-checksum = "04e8453b658fe480c3e70c8ed4e3d3ec33eb74988bd186561b0cc66b85c3bc4b"
++checksum = "a02a88a17e74cadbc8ce77855e1d6c8ad0ab82901a4a9b5046bd01c1c0bd95cd"
+ dependencies = [
+  "cfg-if",
+  "indoc",
+@@ -260,9 +260,9 @@ dependencies = [
+ 
+ [[package]]
+ name = "pyo3-build-config"
+-version = "0.20.0"
++version = "0.21.0"
+ source = "registry+https://github.com/rust-lang/crates.io-index"
+-checksum = "a96fe70b176a89cff78f2fa7b3c930081e163d5379b4dcdf993e3ae29ca662e5"
++checksum = "a5eb0b6ecba38961f6f4bd6cd5906dfab3cd426ff37b2eed5771006aa31656f1"
+ dependencies = [
+  "once_cell",
+  "target-lexicon",
+@@ -270,9 +270,9 @@ dependencies = [
+ 
+ [[package]]
+ name = "pyo3-ffi"
+-version = "0.20.0"
++version = "0.21.0"
+ source = "registry+https://github.com/rust-lang/crates.io-index"
+-checksum = "214929900fd25e6604661ed9cf349727c8920d47deff196c4e28165a6ef2a96b"
++checksum = "ba8a6e48a29b5d22e4fdaf132d8ba8d3203ee9f06362d48f244346902a594ec3"
+ dependencies = [
+  "libc",
+  "pyo3-build-config",
+@@ -280,9 +280,9 @@ dependencies = [
+ 
+ [[package]]
+ name = "pyo3-macros"
+-version = "0.20.0"
++version = "0.21.0"
+ source = "registry+https://github.com/rust-lang/crates.io-index"
+-checksum = "dac53072f717aa1bfa4db832b39de8c875b7c7af4f4a6fe93cdbf9264cf8383b"
++checksum = "4e80493c5965f94a747d0782a607b2328a4eea5391327b152b00e2f3b001cede"
+ dependencies = [
+  "proc-macro2",
+  "pyo3-macros-backend",
+@@ -292,9 +292,9 @@ dependencies = [
+ 
+ [[package]]
+ name = "pyo3-macros-backend"
+-version = "0.20.0"
++version = "0.21.0"
+ source = "registry+https://github.com/rust-lang/crates.io-index"
+-checksum = "7774b5a8282bd4f25f803b1f0d945120be959a36c72e08e7cd031c792fdfd424"
++checksum = "fcd7d86f42004025200e12a6a8119bd878329e6fddef8178eaafa4e4b5906c5b"
+ dependencies = [
+  "heck",
+  "proc-macro2",
-- 
2.48.1

